src/account/account.service.ts

```diff
@@ -37,8 +46,44 @@ export class AccountService {
     constructor(
         private readonly db: Knex,
         private readonly events: EventEmitter,
+        private readonly accountRepository: KnexAccountRepository,
+        private readonly fedifyContextFactory: FedifyContextFactory,
     ) {}
 
+    /**
+     * Get an Account by the ActivityPub ID
+     * If it is not found locally in our database it will be
+     * remotely fetched and stored
+     */
+    async getByApId(id: URL): Promise<Account> {
+        const account = await this.accountRepository.getByApId(id);
+        if (account) {
+            return account;
+        }
+
+        const context = this.fedifyContextFactory.getFedifyContext();
+
+        const documentLoader = await context.getDocumentLoader({
+            handle: 'index',
+        });
+        const potentialActor = await lookupObject(id, { documentLoader });
+
+        if (!isActor(potentialActor)) {
+            throw new URL('Account not found');
+        }
+
+        const data = await mapActorToExternalAccountData(potentialActor);
+
+        await this.createExternalAccount(data);
+
+        const newlyCreatedAccount = await this.accountRepository.getByApId(id);
```

Comment:
How come we can't use the result of `createExternalAccount`?

Reply:
`createExternalAccount` is still using the old `Account` type which is just data and no functionality